src/lib/components/Gallery.svelte

```diff
@@ -1,13 +1,14 @@
 <script lang="ts">
-	import { galleryStore } from "$lib/store"
-	import Image from "$lib/components/Image.svelte"
-
+	import { galleryStore } from '$lib/store'
+	import Image from '$lib/components/Image.svelte'
 </script>
 
 {#if $galleryStore.images?.length}
-<div class="grid grid-cols-4 md:grid-cols-5 xl:grid-cols-7 gap-3 mt-10">
-	{#each $galleryStore.images as image, i}
-		<Image bind:image />
-	{/each}
-</div>
+	<div
+		class="grid grid-cols-2 md:grid-cols-4 xl:grid-cols-6 px-2 gap-3 mt-10 max-h-[520px] overflow-auto"
+	>
```

Comment:
Again this max height feels arbitrary, on very large screens I assume this would look incorrect